### DISCUSSION TITLE
How to persistent auth state in the right way?

### DISCUSSION THREAD

**Discussion by kterto** [serverpod/serverpod#2271] [Q&A] [answered]
I've read the serverpod documentation and the blog post linked there about Authentication. And Things are working, I can sign up and sign in. But one common feature we use in mobile is to persist the auth state, so that the user can access the app without having to sign in again, as long as his token is still valid. I've tryed to use in the flutter side the `SessionManager().signedInUser` but after a hot reload, it returns null, even though the server logs that there is an user signed in. So isntead of that I've created an Endpoint and used the Session object, on the server side, to get the signedIn userId and than retrieved user data from UserInfo table on the DB. But it seems strange to me, the SessionManager in the flutter side would than became not useful at all proceeding in this way. So what is the right way of retrieving user data and use sessionManager in a new app run? 
ps.: I'm suspicious about the need of registering manually the user in the SessionManager, the same way we do after a signIn, but I had some trouble to retrieve the required data, say authenticationKey, and authenticationKeyId.

**Comment by Aygistov** [comment, marked as answer]
Hi!
You can try **sessionManager.initialize()** to initialize the session with storage data.

**Reply by vguardiola** [reply]
Yes, and it does not work

**Comment by vguardiola** [comment]
I've the same problem. I try a lot of things, but I don't find the correct key. In my case I initialize my storage service, get the token save on it and put to the auth manager before create the client, I try to set it after create the client, and after that I initialize the session Manager but ever the request to status.getUserInfo return null and on the server logs the user is null

METHOD         serverpod_auth.status.getUserInfo user=null, queries=0, duration=0µs

Someone have an implementation that's works properly and can share the code
